core/src/peer_id.rs

```diff
@@ -69,17 +70,16 @@ impl PeerId {
         // will switch to not hashing the key (i.e. the correct behaviour).
         // In other words, rust-libp2p 0.16 is compatible with all versions of rust-libp2p.
         // Rust-libp2p 0.12 and below is **NOT** compatible with rust-libp2p 0.17 and above.
-        let (hash_algorithm, canonical_algorithm) = /*if key_enc.len() <= MAX_INLINE_KEY_LENGTH {
+        let (hash_algorithm, canonical_algorithm): (_, Option<Code>) = /*if key_enc.len() <= MAX_INLINE_KEY_LENGTH {
             (multihash::Hash::Identity, Some(multihash::Hash::SHA2256))
         } else {*/
-            (multihash::Hash::SHA2256, None);
+            (Code::Sha2_256, None);
         //};
 
         let canonical = canonical_algorithm.map(|alg|
-            multihash::encode(alg, &key_enc).expect("SHA2256 is always supported"));
+            alg.hasher().expect("canonical hasher exists").digest(&key_enc));
 
-        let multihash = multihash::encode(hash_algorithm, &key_enc)
-            .expect("identity and sha2-256 are always supported by known public key types");
+        let multihash = hash_algorithm.hasher().expect("hasher exists").digest(&key_enc);
```

Comment:
@twittner should I also change this message back to "identity and sha2-256 are always supported by known public key types"?

Reply:
I would go with "identity and sha2-256 are always supported".